### PR TITLE
.Net: OpenAI + AzureOpenAI Connector SDK updates

### LINKED ISSUE #8525: .Net: Bug: HTTP 404 - POST /chat/completions
**Describe the bug**
Exception thrown: 'Microsoft.SemanticKernel.HttpOperationException' in mscorlib.dll
Exception thrown: 'System.AggregatException' in mscorlib.dll
HTTP 404 (invalid_request_error: unknown_url)

Unknown request URL: POST /chat/completions. Please check the URL for typos, or see the docs at https://platform.openai.com/docs/api-reference/.
HTTP 404 (invalid_request_error: unknown_url)

**To Reproduce**
```
var builder = Kernel.CreateBuilder();
builder.AddOpenAIChatCompletion("gpt-4o", OPENAI_APIKEY);
var _kernel = builder.Build();

var service = _kernel.GetRequiredService<IChatCompletionService>();

var settings = new OpenAIPromptExecutionSettings
{
    Temperature = 1.05F,
    TopP = 1F,
    MaxTokens = 4096,
    ResponseFormat = "json_object"
};

var chat = new ChatHistory();
var ct = new CancellationTokenSource();
chat.AddUserMessage("tell me something good");
var content = await service.GetChatMessageContentsAsync(chat, settings, _kernel, ct.Token);
```

**Expected behavior**
'content' variable contains the message result.

**Platform**
 - OS: Windows 11
 - IDE: Visual Studio 2022
 - Language: C#
 - Source: Microsoft.SemanticKernel v 1.18.2

**Additional context**
Was working, then suddenly started throwing this exception.

### PR DESCRIPTION
**Pull request by RogerBarreto**
### Motivation and Context

- Bumping SDK versions and small adjustments
- [Latest OpenAI beta.11 reverted capability to use more customized Uri](https://github.com/openai/openai-dotnet/pull/192), which ensure SK previous behavior, tests were brought back as the previous experience.

Fixes #8525

### CODE REVIEW

**Review by westey-m — APPROVED** [review]

File: dotnet/src/Connectors/Connectors.OpenAI/Services/OpenAIChatCompletionService.cs [hunk 1]
```diff
@@ -71,17 +71,25 @@ public OpenAIChatCompletionService(
...
             else
             {
-                internalClientEndpoint = providedEndpoint;
+                // As OpenAI Client automatically adds the chatcompletions endpoint, we remove it to avoid duplication.
+                const string PathAndQueryPattern = "/chat/completions";
+                var providedEndpointText = providedEndpoint.ToString();
+                int index = providedEndpointText.IndexOf(PathAndQueryPattern, StringComparison.OrdinalIgnoreCase);
+                if (index >= 0)
+                {
+                    internalClientEndpoint = new Uri($"{providedEndpointText.Substring(0, index)}{providedEndpointText.Substring(index + PathAndQueryPattern.Length)}");
```
Comment (westey-m) [inline dotnet/src/Connectors/Connectors.OpenAI/Services/OpenAIChatCompletionService.cs:87]: Should we consider the case where for some reason the person passes in `https://api.openai.com/v1/chat/completions/somethingelse`.
E.g. we could just not do the replacement in this case...

File: dotnet/src/Connectors/Connectors.OpenAI/Core/ClientCore.TextToAudio.cs [hunk 2]
```diff
@@ -58,15 +58,17 @@ private static GeneratedSpeechVoice GetGeneratedSpeechVoice(string? voice)
...
-            "WAV" => (GeneratedSpeechFormat.Wav, "audio/wav"),
-            "MP3" => (GeneratedSpeechFormat.Mp3, "audio/mpeg"),
-            "OPUS" => (GeneratedSpeechFormat.Opus, "audio/opus"),
-            "FLAC" => (GeneratedSpeechFormat.Flac, "audio/flac"),
-            "AAC" => (GeneratedSpeechFormat.Aac, "audio/aac"),
-            "PCM" => (GeneratedSpeechFormat.Pcm, "audio/l16"),
-            null => (null, null),
-            _ => throw new NotSupportedException($"The format '{format}' is not supported.")
-        };
+            case "WAV": return (GeneratedSpeechFormat.Wav, "audio/wav");
```
Comment (dmytrostruk) [inline dotnet/src/Connectors/Connectors.OpenAI/Core/ClientCore.TextToAudio.cs:64]: Just wondering what is the purpose of this change?

File: dotnet/src/Connectors/Connectors.OpenAI/Core/ClientCore.cs [hunk 3]
```diff
@@ -36,7 +36,7 @@ internal partial class ClientCore
     /// <summary>
     /// Default OpenAI API endpoint.
     /// </summary>
-    private const string OpenAIEndpoint = "https://api.openai.com/";
+    private const string OpenAIV1Endpoint = "https://api.openai.com/v1";
```
Comment (dmytrostruk) [inline dotnet/src/Connectors/Connectors.OpenAI/Core/ClientCore.cs:39]: Is there a possibility of behavioral breaking change?

File: dotnet/src/Connectors/Connectors.OpenAI/Core/ClientCore.TextToAudio.cs (hunk 2 above)
Comment (RogerBarret0) [reply dotnet/src/Connectors/Connectors.OpenAI/Core/ClientCore.TextToAudio.cs:64]: The latest change in `OpenAI SDK` breaks usage with switch expression in the line 69 (ArgumentNull Exception), it seems that it creates the `enum` even when I specify (null, null). 
Fallbacking into the classic `switch` statement solves the problem.

File: dotnet/src/Connectors/Connectors.OpenAI/Core/ClientCore.cs (hunk 3 above)
Comment (RogerBarret0) [reply dotnet/src/Connectors/Connectors.OpenAI/Core/ClientCore.cs:39]: Actually this was the original behavior of V1, I had to change it after `OpenAI SDK beta9` which introduced the `v1` enforcement, but now in the `beta.11` they reverted which allowed me to fallback to the previous logic. (your seeing here)

File: dotnet/src/Connectors/Connectors.OpenAI/Services/OpenAIChatCompletionService.cs [hunk 4]
```diff
@@ -71,17 +71,25 @@ public OpenAIChatCompletionService(
...
+            // If the provided endpoint does not provide a path, we add a version to the base path for compatibility
+            if (providedEndpoint.PathAndQuery.Length == 0 || providedEndpoint.PathAndQuery == "/")
             {
-                internalClientEndpoint = new Uri($"{providedEndpointText.Substring(0, index)}{providedEndpointText.Substring(index + PathAndQueryPattern.Length)}");
+                internalClientEndpoint = new Uri(providedEndpoint, "/v1/");
             }
             else
             {
-                internalClientEndpoint = providedEndpoint;
+                // As OpenAI Client automatically adds the chatcompletions endpoint, we remove it to avoid duplication.
```
Comment (markwallace-microsoft) [inline dotnet/src/Connectors/Connectors.OpenAI/Services/OpenAIChatCompletionService.cs:81]: Have we always done this?

Correctly a bad endpoint leaves us open to having to continue to do this as things change and also do this consistently across all of our connectors.

Consider using the provided endpoint and expecting developers to get this correct.

**Review by markwallace-microsoft — APPROVED** [review]
